tests/all/test_attributes.rs

```diff
@@ -40,6 +40,36 @@ fn test_string_attributes() {
     assert_eq!(string_attribute.get_string_value().to_str(), Ok("my_val"));
 }
 
+#[llvm_versions(12.0..=latest)]
+#[test]
+fn test_type_attribute() {
+    use inkwell::types::AnyType;
+
+    let context = Context::create();
+    let kind_id = Attribute::get_named_enum_kind_id("sret");
+    let any_type = context.i32_type().as_any_type_enum();
+    let type_attribute = context.create_type_attribute(
+        kind_id,
+        any_type,
+    );
+
+    assert!(type_attribute.is_type());
```

Comment:
Would be good to add asserts that it's not a string or enum attr

Reply:
Tested and confirmed they are mutually exclusive.